Testing/Unit/sitkImageIOTests.cxx

```diff
@@ -563,23 +563,32 @@ TEST(IO, ImageFileReader_ImageInformation )
   EXPECT_ANY_THROW( reader.GetMetaData("nothing") );
 
 
+  sitk::Image image = sitk::ReadImage(dicomFile1);
   reader.ReadImageInformation();
 
   EXPECT_EQ(reader.GetPixelID(), sitk::sitkInt16);
+  EXPECT_EQ(reader.GetPixelID(), image.GetPixelID());
   EXPECT_EQ(reader.GetPixelIDValue(), sitk::sitkInt16);
+  EXPECT_EQ(reader.GetPixelIDValue(), image.GetPixelIDValue());
   EXPECT_EQ(reader.GetDimension(), 3u);
+  EXPECT_EQ(reader.GetDimension(), image.GetDimension());
   EXPECT_EQ(reader.GetNumberOfComponents(), 1u);
+  EXPECT_EQ(reader.GetNumberOfComponents(), image.GetNumberOfComponentsPerPixel());
   EXPECT_VECTOR_DOUBLE_NEAR(reader.GetOrigin(), v3(-112, -21.687999, 126.894000 ), 1e-6);
+  EXPECT_VECTOR_DOUBLE_NEAR(reader.GetOrigin(), image.GetOrigin(), 1e-6);
   EXPECT_VECTOR_DOUBLE_NEAR(reader.GetSpacing(), v3(0.859375, 0.8593899, 1.600000 ), 1e-6);
-  EXPECT_VECTOR_DOUBLE_NEAR(reader.GetDirection(), v9(1, 0, 0,
-                                                      0, 0.46665081166793676, -0.88444164305490258,
-                                                      -0, 0.88444164305490258, 0.46665081166793676 ), 1e-8);
+  EXPECT_VECTOR_DOUBLE_NEAR(reader.GetSpacing(), image.GetSpacing(), 1e-6);
+  EXPECT_VECTOR_DOUBLE_NEAR(reader.GetDirection(), v9(1, 0, -0,
```

Comment:
Out of curiosity, does the sign of the zero (-0 vs. 0) matter here? I don't expect it to. Is this related to how EXPECT_VECTOR_DOUBLE_NEAR is implemented?

Reply:
Technically IEEE 754 floating point representation has two zeros for the sign. This baseline value is now how it was printed. It makes now numeric difference, but visually indicate the proper matrix orientation.